index.d.ts

```diff
@@ -25,4 +27,17 @@ export class AbortError extends Error {
  * 	throttled(i).then(console.log);
  * }
  */
-export default function <T extends Function>(input: T, limit: number, interval: number): T;
+export default function<TArguments extends any[], TReturn>(
+	fn: (...arguments: TArguments) => PromiseLike<TReturn> | TReturn,
```

Comment:
`fn` => `function`

Reply:
This doesn't work, TS seems to forbid the `function` a keyword here.